mindmeld/components/question_answerer.py

```diff
@@ -823,194 +1097,159 @@ def _update_data_type(self, value):
+                return []
+
             # new: https://github.com/cisco/mindmeld/issues/291
             #   making 'value' into a list for cname and whitelist conversion.
             #   If more than one items in 'value', all items after first one go into whitelist
             entity_map = {
                 "entities": [
                     {
                         "id": _id,
-                        "cname": self._get_resolvers_cname(value),
-                        "whitelist": self._get_resolvers_whitelist(value)
+                        "cname": self.get_resolvers_cname(value),
+                        "whitelist": _get_resolvers_whitelist(value)
                     } for _id, value in id2value.items()
                 ]
             }
             return entity_map
 
-        def update_resource(self, id2value, has_text_resolver, has_embedding_resolver,
-                            resolver_settings, lazy_clean=False,
-                            app_path=NATIVE_QUESTION_ANSWERER_INDICES_CACHE_PATH,
-                            processor_type="keyword"):
+    class FieldResource(FieldResourceProcessor):
+        """
+        An object encapsulating all resources necessary for search/filter/sort-ing on any field in
+        the Knowledge Base. This class should only be used as part of `Indices` class and not in
+        isolation.
+
+        This class currently supports:
+        - location strings,
+        - date strings,
+        - boolean,
+        - number,
+        - strings, and
+        - list of strings.
+
+        Any other data type (eg. dictionary type) is currently not supported and is marked as an
+        'unknown' data type. Such unknown data types fields do not have any associated resolvers.
+        """
+
+        def __init__(self, index_name, field_name):
+
+            # details to establish a scoped field name
+            self.index_name = index_name
+            self.field_name = field_name
+
+            # vars that contain data of the field
+            self.data_type = None
+            self.id2value = {}  # TODO: duplicate data also exist in resolver object if created
+            self.hash = None  # to identify any data changes before build resolver
+
+            # details to create any required resolvers
+            self.processor_type = None
+            self.has_text_resolver = None
+            self.has_embedding_resolver = None
+
+            # required resolvers
+            self._text_resolver = None  # an entity resolver if string type data
+            self._embedding_resolver = None  # an embedding based entity resolver
+
+        def __repr__(self):
+            return f"{self.__class__.__name__} " \
+                   f"field_name: {self.field_name} " \
+                   f"data_type: {self.data_type} " \
+                   f"has_text_resolver: {self.has_text_resolver} " \
+                   f"has_embedding_resolver: {self.has_embedding_resolver}"
+
+        def update_resource(
```

Comment:
maybe we can consider having another method `load_resource()` which can be used in `build_search()` in question answerer class and also here based on hash?